src/android/StatusBar.java

```diff
@@ -169,21 +169,20 @@ public boolean execute(final String action, final CordovaArgs args, final Callba
     }
 
     private void setStatusBarBackgroundColor(final String colorPref) {
-        if (colorPref != null && !colorPref.isEmpty()) {
-            final Window window = cordova.getActivity().getWindow();
-            // Method and constants not available on all SDKs but we want to be able to compile this code with any SDK
-            window.clearFlags(0x04000000); // SDK 19: WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
-            window.addFlags(0x80000000); // SDK 21: WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
-            try {
-                // Using reflection makes sure any 5.0+ device will work without having to compile with SDK level 21
-                window.getClass().getMethod("setStatusBarColor", int.class).invoke(window, Color.parseColor(colorPref));
-            } catch (IllegalArgumentException ignore) {
-                LOG.e(TAG, "Invalid hexString argument, use f.i. '#999999'");
-            } catch (Exception ignore) {
-                // this should not happen, only in case Android removes this method in a version > 21
-                LOG.w(TAG, "Method window.setStatusBarColor not found for SDK level " + Build.VERSION.SDK_INT);
-            }
+        if (colorPref.isEmpty()) return;
```

Comment:
I'm not sure if this is sufficient (and not sure if the old implementation was sufficient either).

[JSONArray.getString](https://developer.android.com/reference/org/json/JSONArray#getString(int)) will throw a `JSONException` if the value does not exists, which is caught on line 140. But does it throw if the value contains a JS Null (e.g. `JSONObject.NULL`)?

I suppose the docs does say:

> Returns the value at index if it exists, coercing it if necessary.

So I guess you'll may receive `"null"` as a value, which will be presumably caught by the `IllegalArgumentException` catch.

Reply:
I guess if you pass in `null` from the JS side, you'll get a JS exception because we attempt to use `charAt` on the expected string.

```
Uncaught TypeError: Cannot read properties of null (reading 'charAt')
    at Object.backgroundColorByHexString (statusbar.js:76:23)
    at <anonymous>:1:11
```

Which is fine, my main concern was a potential for a fatal crash.